src/common/BaseLightAccount.sol

```diff
@@ -0,0 +1,139 @@
+        _onlyOwnerOrEntryPoint();
+        if (dest.length != func.length || dest.length != value.length) {
+            revert ArrayLengthMismatch();
+        }
+        uint256 length = dest.length;
+        for (uint256 i = 0; i < length; ++i) {
+            _call(dest[i], value[i], func[i]);
+        }
+    }
+
+    /// @notice Deposit more funds for this account in the entry point.
+    function addDeposit() public payable {
+        entryPoint().depositTo{value: msg.value}(address(this));
+    }
+
+    /// @notice Withdraw value from the account's deposit.
+    /// @param withdrawAddress Target to send to.
+    /// @param amount Amount to withdraw.
+    function withdrawDepositTo(address payable withdrawAddress, uint256 amount) public onlyOwner {
+        entryPoint().withdrawTo(withdrawAddress, amount);
+    }
+
+    /// @inheritdoc BaseAccount
+    function entryPoint() public view virtual override returns (IEntryPoint) {
+        return _ENTRY_POINT;
+    }
+
+    /// @notice Check current account deposit in the entry point.
+    /// @return The current account deposit.
+    function getDeposit() public view returns (uint256) {
+        return entryPoint().balanceOf(address(this));
+    }
+
+    /// @inheritdoc IERC1271
+    /// @dev Must override to support ERC-1271 signature validation.
+    function isValidSignature(bytes32 hash, bytes memory signature) public view virtual override returns (bytes4);
+
+    /// @dev Must override to support user op validation.
+    function _validateSignature(PackedUserOperation calldata userOp, bytes32 userOpHash)
+        internal
+        virtual
+        override
+        returns (uint256);
+
+    /// @dev Must override to allow calls to protected functions.
+    function _isFromOwner() internal view virtual returns (bool);
+
+    function _isFromEntryPoint() internal view returns (bool) {
+        return msg.sender == address(entryPoint());
+    }
+
+    /// @dev Revert if the caller is not an owner or the account itself (when redirected through `execute`).
+    function _onlyOwner() internal view {
+        if (msg.sender != address(this) && !_isFromOwner()) {
+            revert NotAuthorized(msg.sender);
+        }
+    }
+
+    /// @dev Require that the call is from the entry point or an owner.
+    function _onlyOwnerOrEntryPoint() internal view {
+        if (!_isFromEntryPoint() && !_isFromOwner()) {
+            revert NotAuthorized(msg.sender);
+        }
+    }
+
+    function _call(address target, uint256 value, bytes memory data) internal {
+        (bool success, bytes memory result) = target.call{value: value}(data);
+        if (!success) {
+            assembly {
+                revert(add(result, 32), mload(result))
+            }
+        }
+    }
+
+    function _authorizeUpgrade(address newImplementation) internal view override {
+        (newImplementation);
+        _onlyOwner();
```

Comment:
Could we make this use `_onlyOwnerOrEntryPoint`?

Reply:
Yup, I'll address the modifier changes in another PR.